app/src/main/java/com/example/playlistmaker/data/NetworkClient.kt

```diff
@@ -0,0 +1,7 @@
+package com.example.playlistmaker.data
+
+import com.example.playlistmaker.data.dto.Responce
+
+interface NetworkClient {
+    fun doRequest(dto: Any): Responce
```

Comment:
```suggestion
    fun searchTracks(request: TrackRequest): Responce
```
Для каждого нового запроса здесь будет создаваться новая функция, а не использоваться универсальная `doRequest`